dspace-api/src/main/java/org/dspace/app/canvasdimension/IIIFApiQueryServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.canvasdimension;
+
+import static org.dspace.app.canvasdimension.Util.checkDimensions;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.dspace.app.canvasdimension.service.IIIFApiQueryService;
+import org.dspace.content.Bitstream;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+/**
+ * Queries the configured IIIF server for image dimensions. Used for
+ * formats that cannot be easily read using ImageIO (jpeg 2000).
+ *
+ * @author Michael Spalti mspalti@willamette.edu
+ */
+public class IIIFApiQueryServiceImpl implements IIIFApiQueryService, InitializingBean {
+
+    @Autowired()
+    protected ConfigurationService configurationService;
+
+    String iiifImageServer;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        iiifImageServer = configurationService.getProperty("iiif.image.server");
+    }
+
+    @Override
+    public int[] getImageDimensions(Bitstream bitstream) {
+        return getIiifImageDimensions(bitstream);
+    }
+
+    /**
+     * Retrieves image dimensions from the image server (IIIF Image API v.2.1.1).
+     * @param bitstream the bitstream DSO
+     * @return image dimensions
+     */
+    private int[] getIiifImageDimensions(Bitstream bitstream) {
+        int[] arr = new int[2];
+        String path = iiifImageServer + bitstream.getID() + "/info.json";
```

Comment:
as in https://github.com/DSpace/DSpace/pull/8047/files#r783791480
would be possible to delegate the generation of the info.json url to the `ImageContentService` ?

Reply:
I added the method to `org.dspace.iiif.IIIFSharedUtils`.  On the IIIF API side we can use this method from within `ImageContentService`.